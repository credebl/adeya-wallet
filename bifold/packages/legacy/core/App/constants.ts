import { PINValidationRules } from './types/security'

export const defaultLanguage = 'en'
export const dateIntFormat = 'YYYYMMDD'

const lengthOfhiddenAttributes = 10
const unicodeForBulletCharacter = '\u2022'
export const hiddenFieldValue = Array(lengthOfhiddenAttributes).fill(unicodeForBulletCharacter).join('')
// Used to property prefix TestIDs so they can be looked up
// by on-device automated testing systems like SauceLabs.
export const testIdPrefix = 'com.ariesbifold:id/'

export enum LocalStorageKeys {
  Onboarding = 'OnboardingState',
  LoginAttempts = 'LoginAttempts',
  // FIXME: Once hooks are updated this should no longer be necessary
  RevokedCredentials = 'RevokedCredentials',
  RevokedCredentialsMessageDismissed = 'RevokedCredentialsMessageDismissed',
  Preferences = 'PreferencesState',
  Tours = 'ToursState',
}

export enum KeychainServices {
  Salt = 'secret.wallet.salt',
  Key = 'secret.wallet.key',
}

export enum EventTypes {
  ERROR_ADDED = 'ErrorAdded',
  ERROR_REMOVED = 'ErrorRemoved',
  BIOMETRY_UPDATE = 'BiometryUpdate',
  BIOMETRY_ERROR = 'BiometryError',
}

export const second = 1000
export const minute = 60000
export const hour = 3600000

// wallet timeout of 5 minutes, 0 means the wallet never locks due to inactivity
export const walletTimeout = minute * 5

/* lockout attempt rules: The base rules apply the lockout at a specified number of incorrect attempts,
 and the threshold rules apply the lockout penalty to each attempt after the threshold that falls on the attemptIncrement.
 (In this case the threshold rule applies to every 5th incorrect login after 20)
5 incorrect => 1 minute lockout
10 incorrect => 10 minute lockout
15 incorrect => 1 hour lockout
20, 25, 30, etc incorrect => 1 day lockout
*/
export const attemptLockoutBaseRules: Record<number, number | undefined> = {
  5: minute,
  10: 10 * minute,
  15: hour,
}
export const attemptLockoutThresholdRules = {
  attemptThreshold: 20,
  attemptIncrement: 5,
  attemptPenalty: 24 * hour,
}

export const walletId = 'walletId'

export const minPINLength = 6

export const PINRules: PINValidationRules = {
  only_numbers: true,
  min_length: 6,
  max_length: 6,
  no_repeated_numbers: false,
  no_repetition_of_the_two_same_numbers: false,
  no_series_of_numbers: false,
  no_even_or_odd_series_of_numbers: false,
  no_cross_pattern: false,
}

export const domain = 'didcomm://invite'

export const tourMargin = 25
