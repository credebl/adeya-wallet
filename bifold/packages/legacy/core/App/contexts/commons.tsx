import React, { createContext, useContext } from 'react'

import { AppConsoleLogger, LogLevel } from '../services/logger'

export interface AppCommonsContext {
  log: (message: string, logLevel: Exclude<LogLevel, LogLevel.off>) => void
}

export const AppCommons = createContext<AppCommonsContext>(null as unknown as AppCommonsContext)

export const CommonUtilProvider: React.FC = ({ children }) => {
  const logger = new AppConsoleLogger(LogLevel.test)
  const log = async (messaage: string, logLevel: Exclude<LogLevel, LogLevel.off>) => {
    logger.log(logLevel, messaage)
    //TODO: do some more logic like collecting errors for later logging and investigation
  }

  return (
    <AppCommons.Provider
      value={{
        log: log,
      }}
    >
      {children}
    </AppCommons.Provider>
  )
}

export const useCommons = () => {
  const commonsContext = useContext(AppCommons)
  if (!commonsContext) {
    throw new Error('commonsContext must be used within a CommonUtilProvider')
  }
  return commonsContext
}
