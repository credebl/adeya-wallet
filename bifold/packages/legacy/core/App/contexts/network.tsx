import { NetInfoStateType, useNetInfo } from '@react-native-community/netinfo'
import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import NetInfoModal from '../components/modals/NetInfoModal'
import { fetchLedgerNodes, canConnectToLedgerNode } from '../utils/ledger'

export interface NetworkContext {
  silentAssertConnectedNetwork: () => boolean
  assertConnectedNetwork: () => boolean
  displayNetInfoModal: () => void
  hideNetInfoModal: () => void
  assertLedgerConnectivity: () => Promise<boolean>
}

export const NetworkContext = createContext<NetworkContext>(null as unknown as NetworkContext)

export const NetworkProvider: React.FC = ({ children }) => {
  const netInfo = useNetInfo()
  const [isNetInfoModalDisplayed, setIsNetInfoModalDisplayed] = useState<boolean>(false)

  const displayNetInfoModal = () => {
    setIsNetInfoModalDisplayed(true)
  }

  const hideNetInfoModal = () => {
    setIsNetInfoModalDisplayed(false)
  }

  const silentAssertConnectedNetwork = () => {
    return netInfo.isConnected || netInfo.type !== NetInfoStateType.none
  }

  const assertConnectedNetwork = () => {
    const isConnected = silentAssertConnectedNetwork()
    if (!isConnected) {
      displayNetInfoModal()
    }

    return isConnected
  }

  const assertLedgerConnectivity = async (): Promise<boolean> => {
    const nodes = fetchLedgerNodes()

    if (typeof nodes === 'undefined' || nodes.length === 0) {
      return false
    }

    const connections = await Promise.all(nodes.map((n: { host: string; port: number }) => canConnectToLedgerNode(n)))
    return connections.includes(true)
  }

  return (
    <NetworkContext.Provider
      value={{
        silentAssertConnectedNetwork,
        assertConnectedNetwork,
        displayNetInfoModal,
        hideNetInfoModal,
        assertLedgerConnectivity,
      }}
    >
      {children}
      <NetInfoModal visible={isNetInfoModalDisplayed} onSubmit={() => hideNetInfoModal()} />
    </NetworkContext.Provider>
  )
}

export const useNetwork = () => useContext(NetworkContext)
