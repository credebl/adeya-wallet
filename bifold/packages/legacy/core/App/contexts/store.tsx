import React, { createContext, Dispatch, useContext, useReducer } from 'react'

import { State } from '../types/state'

import _defaultReducer, { ReducerAction } from './reducers/store'

type Reducer = <S extends State>(state: S, action: ReducerAction<any>) => S

interface StoreProviderProps {
  initialState?: State
  reducer?: Reducer
}

export const defaultState: State = {
  onboarding: {
    didAgreeToTerms: false,
    didCompleteTutorial: false,
    didCreatePIN: false,
    didConsiderBiometry: false,
  },
  authentication: {
    didAuthenticate: false,
  },
  loginAttempt: {
    loginAttempts: 0,
    servedPenalty: true,
  },
  lockout: {
    displayNotification: false,
  },
  preferences: {
    developerModeEnabled: false,
    biometryPreferencesUpdated: false,
    useBiometry: false,
    useVerifierCapability: false,
    useConnectionInviterCapability: false,
  },
  tours: {
    seenToursPrompt: false,
    enableTours: true,
    seenHomeTour: false,
  },
  deepLink: {
    activeDeepLink: '',
  },
  loading: false,
}

export const StoreContext = createContext<[State, Dispatch<ReducerAction<any>>]>([
  defaultState,
  () => {
    return
  },
])

export const mergeReducers = (a: Reducer, b: Reducer): Reducer => {
  return <S extends State>(state: S, action: ReducerAction<any>): S => {
    return a(b(state, action), action)
  }
}

export const defaultReducer = _defaultReducer

export const StoreProvider: React.FC<StoreProviderProps> = ({ children, initialState, reducer }) => {
  const _reducer = reducer ?? defaultReducer
  const _state = initialState ?? defaultState
  const [state, dispatch] = useReducer(_reducer, _state)

  return <StoreContext.Provider value={[state, dispatch]}>{children}</StoreContext.Provider>
}

export const useStore = <S extends State>(): [S, Dispatch<ReducerAction<any>>] => {
  const context = useContext(StoreContext)

  return context as unknown as [S, Dispatch<ReducerAction<any>>]
}
