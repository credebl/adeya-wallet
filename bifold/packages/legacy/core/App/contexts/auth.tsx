import { agentDependencies } from '@aries-framework/react-native'
import React, { createContext, useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { DeviceEventEmitter } from 'react-native'

import { EventTypes } from '../constants'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import {
  secretForPIN,
  storeWalletSecret,
  loadWalletSecret,
  loadWalletSalt,
  isBiometricsActive,
  wipeWalletKey,
} from '../services/keychain'
import { WalletSecret } from '../types/security'
import { hashPIN } from '../utils/crypto'

export interface AuthContext {
  checkPIN: (PIN: string) => Promise<boolean>
  getWalletCredentials: () => Promise<WalletSecret | undefined>
  removeSavedWalletSecret: () => void
  disableBiometrics: () => Promise<void>
  setPIN: (PIN: string) => Promise<void>
  commitPIN: (useBiometry: boolean) => Promise<boolean>
  isBiometricsActive: () => Promise<boolean>
}

export const AuthContext = createContext<AuthContext>(null as unknown as AuthContext)

export const AuthProvider: React.FC = ({ children }) => {
  const [walletSecret, setWalletSecret] = useState<WalletSecret>()
  const [, dispatch] = useStore()
  const { t } = useTranslation()

  const setPIN = async (PIN: string): Promise<void> => {
    const secret = await secretForPIN(PIN)
    await storeWalletSecret(secret)
  }

  const getWalletCredentials = async (): Promise<WalletSecret | undefined> => {
    if (walletSecret && walletSecret.key) {
      return walletSecret
    }

    const { secret, err } = await loadWalletSecret(
      t('Biometry.UnlockPromptTitle'),
      t('Biometry.UnlockPromptDescription')
    )

    DeviceEventEmitter.emit(EventTypes.BIOMETRY_ERROR, err !== undefined)

    if (!secret) {
      return
    }
    setWalletSecret(secret)

    return secret
  }

  const commitPIN = async (useBiometry: boolean): Promise<boolean> => {
    const secret = await getWalletCredentials()
    if (!secret) {
      return false
    }
    // set did authenticate to true if we can get wallet credentials
    dispatch({
      type: DispatchAction.DID_AUTHENTICATE,
    })
    if (useBiometry) {
      await storeWalletSecret(secret, useBiometry)
    } else {
      // erase wallet key if biometrics is disabled
      await wipeWalletKey(useBiometry)
    }
    return true
  }

  const checkPIN = async (PIN: string): Promise<boolean> => {
    const secret = await loadWalletSalt()

    if (!secret || !secret.salt) {
      return false
    }

    const hash = await hashPIN(PIN, secret.salt)

    try {
      await agentDependencies.indy.openWallet({ id: secret.id }, { key: hash })
      // need full secret in volatile memory in case user wants to fall back to using PIN
      const fullSecret = await secretForPIN(PIN, secret.salt)
      setWalletSecret(fullSecret)
      return true
    } catch (e) {
      return false
    }
  }

  const removeSavedWalletSecret = () => {
    setWalletSecret(undefined)
  }

  const disableBiometrics = async () => {
    await wipeWalletKey(true)
  }

  return (
    <AuthContext.Provider
      value={{
        checkPIN,
        getWalletCredentials,
        removeSavedWalletSecret,
        disableBiometrics,
        commitPIN,
        setPIN,
        isBiometricsActive,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = () => useContext(AuthContext)
