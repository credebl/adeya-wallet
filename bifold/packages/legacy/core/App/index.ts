/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@aries-framework/core'
import AgentProvider from '@aries-framework/react-hooks'

import indyLedgers from '../configs/ledgers/indy'

import * as components from './components'
import Button, { ButtonType } from './components/buttons/Button'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import CredentialCard from './components/misc/CredentialCard'
import InfoBox, { InfoBoxType } from './components/misc/InfoBox'
import ErrorModal from './components/modals/ErrorModal'
import NetInfo from './components/network/NetInfo'
import Record from './components/record/Record'
import InfoTextBox from './components/texts/InfoTextBox'
import { ToastType } from './components/toast/BaseToast'
import toastConfig from './components/toast/ToastConfig'
import { AttachTourStep } from './components/tour/AttachTourStep'
import { homeTourSteps } from './components/tour/HomeTourSteps'
import { TourBox } from './components/tour/TourBox'
import HomeContentView from './components/views/HomeContentView'
import * as contexts from './contexts'
import { AuthProvider } from './contexts/auth'
import { CommonUtilProvider } from './contexts/commons'
import { NetworkProvider } from './contexts/network'
import { useTour } from './contexts/tour/tour-context'
import { TourProvider } from './contexts/tour/tour-provider'
import { defaultConfiguration } from './defaultConfiguration'
import RootStack from './navigators/RootStack'
import AttemptLockout from './screens/AttemptLockout'
import Developer from './screens/Developer'
import OnboardingPages from './screens/OnboardingPages'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UseBiometry from './screens/UseBiometry'
import * as types from './types'

export { LocalStorageKeys } from './constants'
export { initLanguages, initStoredLanguage, translationResources } from './localization'
export { ConfigurationProvider, useConfiguration } from './contexts/configuration'
export { defaultState, mergeReducers, StoreProvider, StoreContext, useStore } from './contexts/store'
export { default as Store, DispatchAction, reducer } from './contexts/reducers/store'
export { Assets as ImageAssets } from './theme'
export { ThemeProvider, useTheme } from './contexts/theme'
export { AnimatedComponentsProvider, useAnimatedComponents } from './contexts/animated-components'
export { ColorPallet } from './theme'
export { animatedComponents } from './animated-components'
export { theme } from './theme'
export { useAuth } from './contexts/auth'
export { NavigationTheme } from './theme'
export { testIdWithKey } from './utils/testable'
export { Screens, Stacks, TabStacks } from './types/navigators'
export { createStyles } from './screens/OnboardingPages'
export { statusBarStyleForColor, StatusBarStyles } from './utils/luminance'
export { BifoldError } from './types/error'
export { EventTypes } from './constants'

export type { AnimatedComponents } from './animated-components'
export type {
  ISVGAssets,
  IFontAttributes,
  IInputAttributes,
  IInputs,
  ITextTheme,
  IBrandColors,
  ISemanticColors,
  INotificationColors,
  IGrayscaleColors,
  IColorPallet,
  IAssets,
  ITheme,
} from './theme'
export type { ConfigurationContext } from './contexts/configuration'
export type { TourStep } from './contexts/tour/tour-context'
export type { GenericFn } from './types/fn'
export type { AuthenticateStackParams } from './types/navigators'
export type { OnboardingStyleSheet }
export type { WalletSecret } from './types/security'
export type { ReducerAction } from './contexts/reducers/store'
export type {
  State,
  Onboarding as OnboardingState,
  LoginAttempt as LoginAttemptState,
  Preferences as PreferencesState,
  Tours as ToursState,
} from './types/state'
export * from '../verifier'

export {
  indyLedgers,
  Agent,
  CommonUtilProvider,
  AgentProvider,
  AuthProvider,
  NetworkProvider,
  TourProvider,
  useTour,
  AttachTourStep,
  TourBox,
  homeTourSteps,
  Button,
  ButtonType,
  CheckBoxRow,
  CredentialCard,
  ErrorModal,
  InfoTextBox,
  InfoBox,
  InfoBoxType,
  ToastType,
  toastConfig,
  RootStack,
  NetInfo,
  OnboardingPages,
  Splash,
  Developer,
  Terms,
  HomeContentView,
  UseBiometry,
  AttemptLockout,
  Record,
  defaultConfiguration,
  types,
  components,
  contexts,
}
