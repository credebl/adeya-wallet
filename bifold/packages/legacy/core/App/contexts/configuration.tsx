import { IndyPoolConfig } from '@aries-framework/core'
import { createContext, ReducerAction, useContext } from 'react'

import { ProofRequestTemplate } from '../../verifier'
import { EmptyListProps } from '../components/misc/EmptyList'
import { RecordProps } from '../components/record/Record'
import OnboardingPages from '../screens/OnboardingPages'
import { ScanProps } from '../screens/Scan'
import { OCABundleResolver } from '../types/oca'
import { PINSecurityParams } from '../types/security'
import { SettingSection } from '../types/settings'

interface NotificationConfiguration {
  component: React.FC
  onCloseAction: (dispatch?: React.Dispatch<ReducerAction<any>>) => void
  title: string
  description: string
  buttonTitle: string
  pageTitle: string
}

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  homeContentView: React.FC
  credentialListHeaderRight: React.FC
  credentialListOptions: React.FC
  credentialEmptyList: React.FC<EmptyListProps>
  developer: React.FC
  OCABundleResolver: OCABundleResolver
  scan: React.FC<ScanProps>
  useBiometry: React.FC
  record: React.FC<RecordProps>
  PINSecurity: PINSecurityParams
  indyLedgers: IndyPoolConfig[]
  settings: SettingSection[]
  customNotification: NotificationConfiguration
  useCustomNotifications: () => { total: number; notifications: any }
  connectionTimerDelay?: number
  autoRedirectConnectionToHome?: boolean
  proofRequestTemplates?: Array<ProofRequestTemplate>
  enableTours?: boolean
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
