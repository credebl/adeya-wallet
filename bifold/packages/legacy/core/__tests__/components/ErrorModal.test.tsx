import { render, fireEvent } from '@testing-library/react-native'
import React, { useContext } from 'react'

import ErrorModal from '../../App/components/modals/ErrorModal'
import * as themeContext from '../../App/contexts/theme' // note we're importing with a * to import all the exports
import { theme } from '../../App/theme'

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: jest.fn(),
}))

const setState = jest.fn()
const state = {
  onboarding: {
    DidAgreeToTerms: false,
    DidCompleteTutorial: false,
    DidCreatePIN: false,
  },
  notifications: {
    ConnectionPending: true,
  },
  error: null,
}

// @ts-ignore
useContext.mockImplementation(() => [state, setState])

describe('ErrorModal Component', () => {
  test('Renders correctly', async () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const tree = render(<ErrorModal />)

    expect(tree).toMatchSnapshot()
    expect(tree.queryByText('Global.Okay')).not.toBeNull()
  })

  test('Dismiss on demand', async () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const tree = render(<ErrorModal />)

    const dismissBtn = await tree.findByText('Global.Okay')
    fireEvent(dismissBtn, 'press')

    expect(tree).toMatchSnapshot()
  })
})
