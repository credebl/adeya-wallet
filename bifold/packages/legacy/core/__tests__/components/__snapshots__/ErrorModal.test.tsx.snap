// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal Component Dismiss on demand 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": 1334,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "borderColor": "#D8292F",
          "borderRadius": 5,
          "borderWidth": 1,
          "minWidth": 700,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingHorizontal": 5,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "marginRight": 10,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#D8292F",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#FFFFFF",
              "flexShrink": 1,
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
          testID="com.ariesbifold:id/HeaderText"
        >
          Error.Unknown
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginLeft": 40,
            "paddingBottom": 5,
            "paddingHorizontal": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "flexShrink": 1,
              "fontSize": 18,
              "fontWeight": "normal",
              "marginVertical": 16,
            }
          }
          testID="com.ariesbifold:id/BodyText"
        >
          Error.Problem
        </Text>
        <View
          accessibilityLabel="Global.Okay"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/Okay"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Okay
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`ErrorModal Component Renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <RNCSafeAreaView
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": 1334,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "borderColor": "#D8292F",
          "borderRadius": 5,
          "borderWidth": 1,
          "minWidth": 700,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingHorizontal": 5,
            "paddingTop": 5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignSelf": "center",
                "marginRight": 10,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#D8292F",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#FFFFFF",
              "flexShrink": 1,
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
          testID="com.ariesbifold:id/HeaderText"
        >
          Error.Unknown
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginLeft": 40,
            "paddingBottom": 5,
            "paddingHorizontal": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "flexShrink": 1,
              "fontSize": 18,
              "fontWeight": "normal",
              "marginVertical": 16,
            }
          }
          testID="com.ariesbifold:id/BodyText"
        >
          Error.Problem
        </Text>
        <View
          accessibilityLabel="Global.Okay"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/Okay"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  },
                  false,
                ]
              }
            >
              Global.Okay
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
